test/cluster/functionality.sh

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+
+# This functionality test ensures that:
+#  - ElasticSearch and Kibana pods have started
+#    successfully
+#  - ElasticSearch is reachable from Kibana
+#  - the Kibana-proxy is working
+#  - the Kibana cert and key are functional
+#  - indices have been successfully created by Fluentd
+#    in ElasticSearch
+#
+# This script expects the following environment
+# variables:
+#  - OAL_{
+#         ELASTICSEACH,
+#         KIBANA
+#         }_COMPONENT: the component labels that
+#    are used to identify application pods
+#  - OAL_ELASTICSEACH_SERVICE: the service under which
+#    ElasticSearch is exposed
+#  - OAL_QUERY_SIZE: the number of messages to query
+#    for per index
+#  - OAL_TEST_IP: the IP address to test forwarding for
+#  - LOG_ADMIN_{
+#               USER
+#               PW
+#               }: credentials for the admin user
+source "${OS_ROOT}/hack/lib/init.sh"
+
+query_size="${OAL_QUERY_SIZE:-"500"}"
+test_ip="${OAL_TEST_IP:-"127.0.0.1"}"
+
+os::test::junit::declare_suite_start "test/cluster/functionality"
+
+os::cmd::expect_success "oc project logging"
+
+# We need to use a name and token for logging checks later,
+# so we have to provision a user with a token for this.
+# TODO: Why is this necessary?
```

Comment:
I suppose we could have a separate test for this.  We need to ensure that username/token auth works.  We need to ensure that admin users can read all indices + the `.operations.*` indices.  We need to ensure that normal users can read the logs for the projects they have access to, and cannot read the logs for other projects or `.operations.*`.